dspace-api/src/main/java/org/dspace/content/BitstreamServiceImpl.java

```diff
@@ -403,7 +403,9 @@ public Bitstream getFirstBitstream(Item item, String bundleName) throws SQLExcep
 
     @Override
     public Bitstream getThumbnail(Context context, Bitstream bitstream) throws SQLException {
-        Pattern pattern = Pattern.compile("^" + bitstream.getName() + ".([^.]+)$");
+        Pattern pattern = Pattern.compile("^" +
+            (bitstream.getName() != null ? Pattern.quote(bitstream.getName()) : bitstream.getName())
```

Comment:
I think we can improve this method by adding a guard at the very first line.
Correct me If I am wrong, @tdonohue, we can skip the whole method if the `bitstream` has a `null` name and just return null.